system/utils.py

```diff
@@ -664,23 +664,36 @@ def get_node_did(node_alias, pool_info=None):
     async def _get_primary():
         req = await ledger.build_get_validator_info_request(trustee_did)
         results = json.loads(await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, req))
+        # get n
+        n = len(results)
+        # calculate f
+        f = (n - 1) // 3
         # remove all timeout entries
         try:
             for i in range(len(results)):
                 results.pop(list(results.keys())[list(results.values()).index('timeout')])
         except ValueError:
             pass
         # remove all not REPLY and empty (not selected) primaries entries
-        results = {key: json.loads(results[key]) for key in results if
-                   (json.loads(results[key])['op'] == 'REPLY')
-                   & (json.loads(results[key])['result']['data']['Node_info']['Replicas_status'][key + ':0']['Primary']
-                      is not None)}
+        results = {
+            key: json.loads(results[key]) for key in results if
+            (json.loads(results[key])['op'] == 'REPLY')
+            &
+            (json.loads(results[key])['result']['data']['Node_info']['Replicas_status'][key + ':0']['Primary']
```

Comment:
This cries to be extracted into helper function like
```
def get_primary_from_validator_info(info, name):
    return info['result']['data']['Node_info']['Replicas_status'][name + ':0']['Primary']
```

Reply:
Ok.